main.py

```diff
@@ -246,18 +265,36 @@ def report():
               'Some values have 0 in them which are from your data or where dates'
               ' were missing and were filled in automatically'
           )
-        from_d = datetime.strptime(data['from_date'], '%Y-%m-%d')
-        to_d = datetime.strptime(data['to_date'], '%Y-%m-%d')
-        event_d = datetime.strptime(data['event_date'], '%Y-%m-%d')
 
-        pre_delta = (event_d - from_d).days
-        post_delta = (to_d - event_d).days
+        try:
+          from_d = df.index.get_loc(data['from_date'])
+          to_d = df.index.get_loc(data['to_date'])
+          event_d = df.index.get_loc(data['event_date'])
+        except KeyError:
```

Comment:
should there be an error log here?

Reply:
No as it is a user error so only to inform the user they have selected a date that does not exist in their data (as I now have enabled them to remove dates).

Reply:
ok. I think it would be better to log it so that to understand user behaviour